api/src/main/java/org/pmiops/workbench/google/CloudBillingClient.java

```diff
@@ -1,9 +1,18 @@
 package org.pmiops.workbench.google;
 
-import com.google.api.services.cloudbilling.model.BillingAccount;
+import com.google.api.services.cloudbilling.model.ProjectBillingInfo;
 import java.io.IOException;
 
 /** Encapsulate Google APIs for interfacing with Google Cloud Billing APIs. */
 public interface CloudBillingClient {
-  BillingAccount getBillingAccount(String billingAccount) throws IOException;
+  /**
+   * Poll GCP project's billing info until matches the expected billing account name.
+   *
+   * @return
+   */
+  ProjectBillingInfo pollUntilBillingAccountLinked(String billingAccountName, String projectId)
```

Comment:
nit: I would probably prefer reversing these parameters. Project feels like the primary resource here.